### PR TITLE
[Tools] Add vit/swin/mit convert weight scripts

### PR DESCRIPTION
**Pull request by xiexinch**
## Motivation

Remove convert keys operation at `init_weight` function, use script instead.

### CODE REVIEW

**Comment by Junjun2016** [comment]
The description or docstring is too short.

**Comment by xiexinch** [comment]
> The description or docstring is too short.

What information should be included?

**Comment by Junjun2016** [comment]
> > The description or docstring is too short.
> 
> What information should be included?

Describe what the script does, such as
### RegNet model to MMDetection
`tools/model_converters/regnet2mmdet.py` convert keys in pycls pretrained RegNet models to MMDetection style.
https://github.com/open-mmlab/mmdetection/blob/master/docs/useful_tools.md#regnet-model-to-mmdetection

Please update the Chinese document at the same time.

File: docs/useful_tools.md [hunk 1]
```diff
@@ -255,6 +255,36 @@ Examples:
   python tools/analyze_logs.py log.json --keys loss --legend loss
   ```
 
+### Model conversion
+
+We provide scripts to convert pretrain models provided by other repos to MMSegmentation format.
+
+#### vit/swin/mit models
```
Comment (Junjun2016) [inline docs/useful_tools.md:262]:
```suggestion
#### ViT Swin MiT Transformer Models
```

File: docs/useful_tools.md [hunk 2]
```diff
@@ -255,6 +255,36 @@ Examples:
...
+
+- timm vit
```
Comment (Junjun2016) [inline docs/useful_tools.md:264]:
```suggestion
- ViT
```

File: docs/useful_tools.md [hunk 3]
```diff
@@ -255,6 +255,36 @@ Examples:
...
+
+  `tools/model_converters/vit2mmseg.py` convert keys in timm pretrained vit models to MMSegmentation style.
+
+  ```shell
+  python tools/model_converters/vit2mmseg.py ${SRC} ${DST}
+  ```
+
+- official swin
```
Comment (Junjun2016) [inline docs/useful_tools.md:272]:
```suggestion
- Swin
```

File: docs/useful_tools.md [hunk 4]
```diff
@@ -255,6 +255,36 @@ Examples:
...
+
+  `tools/model_converters/swin2mmseg.py` convert keys in official pretrained swin models to MMSegmentation style.
+
+  ```shell
+  python tools/model_converters/swin2mmseg.py ${SRC} ${DST}
+  ```
+
+- segformer
```
Comment (Junjun2016) [inline docs/useful_tools.md:280]:
```suggestion
- SegFormer
```

File: tools/model_converters/mit2mmseg.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import argparse
+from collections import OrderedDict
+
+import torch
+from mmcv.runner import _load_checkpoint
+
+
+def mit_convert(ckpt):
```
Comment (Junjun2016) [inline tools/model_converters/mit2mmseg.py:8]:
```suggestion
def convert_mit(ckpt):
```

File: tools/model_converters/swin2mmseg.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
+import argparse
+from collections import OrderedDict
+
+import torch
+from mmcv.runner import _load_checkpoint
+
+
+def swin_convert(ckpt):
```
Comment (Junjun2016) [inline tools/model_converters/swin2mmseg.py:8]:
```suggestion
def convert_swin(ckpt):
```
It's better to use verbs + nouns.

File: tools/model_converters/vit2mmseg.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import argparse
+from collections import OrderedDict
+
+import torch
+from mmcv.runner import _load_checkpoint
+
+
+def vit_convert(ckpt):
```
Comment (Junjun2016) [inline tools/model_converters/vit2mmseg.py:8]:
```suggestion
def convert_vit(ckpt):
```

File: tools/model_converters/vit2mmseg.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+            new_k = new_k.replace('blocks.', 'layers.')
+        else:
+            new_k = k
+        new_ckpt[new_k] = v
+
+    return new_ckpt
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Convert model keys')
```
Comment (Junjun2016) [inline tools/model_converters/vit2mmseg.py:44]: Add more details about this description, should contain convert xxx to xxx.

File: tools/model_converters/swin2mmseg.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+            new_v = v
+            new_k = k
+
+        new_ckpt[new_k] = new_v
+
+    return new_ckpt
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Convert model keys')
```
Comment (Junjun2016) [inline tools/model_converters/swin2mmseg.py:63]: Add more details about this description, should contain convert xxx to xxx.

File: tools/model_converters/mit2mmseg.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            new_v = v
+        else:
+            new_k = k
+            new_v = v
+        new_ckpt[new_k] = new_v
+    return new_ckpt
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Convert model keys')
```
Comment (Junjun2016) [inline tools/model_converters/mit2mmseg.py:56]: Add more details about this description, should contain convert xxx to xxx.

File: docs/useful_tools.md (hunk 1 above)
Comment (MengzhangLI) [reply docs/useful_tools.md:262]: Agree, the first letter should be uppercase letter.

File: docs_zh-CN/useful_tools.md [hunk 11]
```diff
@@ -259,6 +259,30 @@ python tools/analyze_logs.py xxx.log.json [--keys ${KEYS}] [--legend ${LEGEND}]
   python tools/analyze_logs.py log.json --keys loss --legend loss
   ```
 
+### 转换其他仓库的权重
+
+`tools/model_converters/` 提供了若干个预训练权重转换脚本，支持将其他仓库的预训练权重的 key 转换为与 MMSegmentation 相匹配的 key。
+
+#### vit/swin/mit 转换脚本
```
Comment (MengzhangLI) [inline docs_zh-CN/useful_tools.md:266]:
```suggestion
#### ViT Swin MiT Transformer 模型
```

**Review by MengzhangLI — APPROVED** [review]

**Review by MengzhangLI — APPROVED** [review]

File: tools/model_converters/mit2mmseg.py [hunk 12]
```diff
@@ -0,0 +1,77 @@
+import argparse
```
Comment (xvjiarui) [inline tools/model_converters/mit2mmseg.py:1]: We may delete `mit_convert.py`.

File: tools/model_converters/mit2mmseg.py [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+from collections import OrderedDict
+
+import torch
+from mmcv.runner import _load_checkpoint
```
Comment (xvjiarui) [inline tools/model_converters/mit2mmseg.py:5]: Will `torch.load_checkpoint` do?

**Comment by xvjiarui** [comment]
We may delete convert function from ViT/Swin backbones in a new PR. 
And we may add instruction on how to convert model in their README.md.

(It seems that Segformer's readme.md should be renamed as README.md, and model-index is also missing).

File: tools/model_converters/mit2mmseg.py (hunk 13 above)
Comment (xiexinch) [reply tools/model_converters/mit2mmseg.py:5]: I'll use `torch.load()` instead.

File: docs/useful_tools.md [hunk 14]
```diff
@@ -255,6 +255,28 @@ Examples:
...
+  ```shell
+  python tools/model_converters/vit2mmseg.py ${SRC} ${DST}
+  ```
+
+- Swin
+
+  `tools/model_converters/swin2mmseg.py` convert keys in official pretrained swin models to MMSegmentation style.
+
+  ```shell
+  python tools/model_converters/swin2mmseg.py ${SRC} ${DST}
```
Comment (Junjun2016) [inline docs/useful_tools.md:277]: Should also add the document for MiT.
In the MiT checkpoint converting script, change `mit_convert` to `convert_mit`.

File: docs_zh-CN/useful_tools.md [hunk 15]
```diff
@@ -259,6 +259,24 @@ python tools/analyze_logs.py xxx.log.json [--keys ${KEYS}] [--legend ${LEGEND}]
   python tools/analyze_logs.py log.json --keys loss --legend loss
   ```
 
+### 转换其他仓库的权重
+
+`tools/model_converters/` 提供了若干个预训练权重转换脚本，支持将其他仓库的预训练权重的 key 转换为与 MMSegmentation 相匹配的 key。
+
+#### ViT Swin MiT Transformer 模型
+
```
Comment (Junjun2016) [inline docs_zh-CN/useful_tools.md:267]: Should be consistent with the English document.

File: docs/useful_tools.md [hunk 16]
```diff
@@ -277,6 +277,14 @@ We provide scripts to convert pretrain models provided by other repos to MMSegme
   python tools/model_converters/swin2mmseg.py ${SRC} ${DST}
   ```
 
+- Segformer
```
Comment (Junjun2016) [inline docs/useful_tools.md:280]:
```suggestion
- SegFormer
```

File: docs_zh-CN/useful_tools.md [hunk 17]
```diff
@@ -259,6 +259,30 @@ python tools/analyze_logs.py xxx.log.json [--keys ${KEYS}] [--legend ${LEGEND}]
...
+- ViT
+
+  ```shell
+  python tools/model_converters/vit2mmseg.py ${SRC} ${DST}
```
Comment (Junjun2016) [inline docs_zh-CN/useful_tools.md:271]: Should be consistent with the English document.

File: docs_zh-CN/useful_tools.md [hunk 18]
```diff
@@ -259,6 +259,30 @@ python tools/analyze_logs.py xxx.log.json [--keys ${KEYS}] [--legend ${LEGEND}]
...
+  ```
+
+- Swin
+
+  ```shell
+  python tools/model_converters/swin2mmseg.py ${SRC} ${DST}
```
Comment (Junjun2016) [inline docs_zh-CN/useful_tools.md:277]: Should be consistent with the English document.

File: docs_zh-CN/useful_tools.md [hunk 19]
```diff
@@ -259,6 +259,30 @@ python tools/analyze_logs.py xxx.log.json [--keys ${KEYS}] [--legend ${LEGEND}]
...
+  ```
+
+- SegFormer
+
+  ```shell
+  python tools/model_converters/mit2mmseg.py ${SRC} ${DST}
```
Comment (Junjun2016) [inline docs_zh-CN/useful_tools.md:283]: Should be consistent with the English document.

**Review by xvjiarui — APPROVED** [review]
